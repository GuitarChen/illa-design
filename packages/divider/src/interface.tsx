import { HTMLAttributes } from "react"

export type DividerDirection = "vertical" | "horizontal"
export type DividerVariant = "solid" | "dashed" | "dotted" | "double"
export type TextAlign = "start" | "center" | "end"
export type DividerColorScheme =
  | string
  | "gray"
  | "blue"
  | "purple"
  | "red"
  | "green"
  | "yellow"
  | "orange"
  | "cyan"
  | "white"
  | "techPink"
  | "techPurple"

export interface DividerProps extends HTMLAttributes<HTMLDivElement> {
  direction?: DividerDirection
  variant?: DividerVariant
  text?: string
  textAlign?: TextAlign
  textSize?: string
  colorScheme?: DividerColorScheme
}
