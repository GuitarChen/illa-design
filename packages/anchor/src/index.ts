export * from "./anchor"
export * from "./interface"
