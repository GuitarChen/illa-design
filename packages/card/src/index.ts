export * from "./card"
export * from "./interface"
export * from "./card-grid"
export * from "./meta"
