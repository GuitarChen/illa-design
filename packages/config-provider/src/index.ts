export * from "./config-provider"
export * from "./interface"
export * from "./config-provider-context"
export * from "./config"

// locale
export * from "./locale/en-US"
export * from "./locale/zh-CN"
export * from "./locale/def"
