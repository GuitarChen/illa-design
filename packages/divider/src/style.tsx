import { DividerColorScheme, DividerVariant } from "./interface"
import { css, SerializedStyles } from "@emotion/react"
import { globalColor, illaPrefix } from "@illa-design/theme"

const innerColor = [
  "white",
  "blackAlpha",
  "gray",
  "red",
  "orange",
  "yellow",
  "green",
  "blue",
  "cyan",
  "purple",
  "grayBlue",
  "techPurple",
  "techPink",
]

export function applyDividerContainerHorizontal(
  variant: DividerVariant,
): SerializedStyles {
  return css`
    vertical-align: middle;
    border-style: ${variant};
    border-width: 0 0 ${variant == "double" ? "3px" : "1px"} 0;
    width: 100%;
  `
}

export function applyDividerContainerVertical(
  variant: DividerVariant,
): SerializedStyles {
  return css`
    display: inline-flex;
    vertical-align: middle;
    border-width: 0 0 0 ${variant == "double" ? "3px" : "1px"};
    border-image: initial;
    border-style: ${variant};
    height: 1em;
  `
}

export function applyDividerWithTextContainerStyle(
  color?: DividerColorScheme,
  textSize?: string,
): SerializedStyles {
  const _color = color && innerColor.indexOf(color) > -1 ? color : "grayBlue"
  return css`
    display: flex;
    width: 100%;
    align-items: center;
    color: ${globalColor(`--${illaPrefix}-${_color}-08`)};
    font-size: ${textSize};
    span {
      padding: 0 16px;
    }
  `
}

export function applyDividerStyle(
  variant: DividerVariant,
  full?: boolean,
): SerializedStyles {
  return css`
    flex-grow: ${full ? 1 : 0};
    flex-basis: 24px;
    ${applyDividerContainerHorizontal(variant)}
  `
}
