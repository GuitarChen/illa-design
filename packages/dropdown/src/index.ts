export * from "./dropdown"
export * from "./interface"
