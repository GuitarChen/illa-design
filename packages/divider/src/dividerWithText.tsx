import { forwardRef } from "react"
import { DividerProps } from "./interface"
import { applyDividerStyle, applyDividerWithTextContainerStyle } from "./style"

export const DividerWithText = forwardRef<HTMLDivElement, DividerProps>(
  (props, ref) => {
    const {
      variant = "solid",
      text,
      textAlign = "center",
      textSize = "14px",
      colorScheme,
      ...otherProps
    } = props

    return (
      <div
        css={applyDividerWithTextContainerStyle(colorScheme, textSize)}
        ref={ref}
        {...otherProps}
      >
        <div css={applyDividerStyle(variant, textAlign !== "start")} />
        <span>{text}</span>
        <div css={applyDividerStyle(variant, textAlign !== "end")} />
      </div>
    )
  },
)

DividerWithText.displayName = "DividerWithText"
