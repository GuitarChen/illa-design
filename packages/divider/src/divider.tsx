import { forwardRef, useMemo } from "react"
import { SerializedStyles } from "@emotion/react"
import { DividerProps } from "./interface"
import {
  applyDividerContainerHorizontal,
  applyDividerContainerVertical,
  applyDividerWithTextContainerStyle,
} from "./style"
import { DividerWithText } from "./dividerWithText"

export const Divider = forwardRef<HTMLDivElement, DividerProps>(
  (props, ref) => {
    const {
      direction = "horizontal",
      variant = "solid",
      text,
      colorScheme,
      ...otherProps
    } = props

    let dividerCss: SerializedStyles
    switch (direction) {
      case "vertical":
        dividerCss = applyDividerContainerVertical(variant)
        break
      case "horizontal":
        dividerCss = applyDividerContainerHorizontal(variant)
        break
    }

    return (
      <div css={applyDividerWithTextContainerStyle(colorScheme)}>
        {text && text?.length > 0 ? (
          <DividerWithText ref={ref} {...props} />
        ) : (
          <div css={dividerCss} ref={ref} {...otherProps} />
        )}
      </div>
    )
  },
)

Divider.displayName = "Divider"
