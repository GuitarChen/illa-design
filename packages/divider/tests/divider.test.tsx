import { render } from "@testing-library/react"
import { Divider } from "../src"

test("Divider renders default", () => {
  const { asFragment } = render(<Divider />)
  expect(asFragment()).toMatchSnapshot()
})

test("Divider renders horizontal", () => {
  const { asFragment } = render(
    <Divider direction="horizontal" variant="solid" />,
  )
  expect(asFragment()).toMatchSnapshot()
})

test("Divider renders vertical", () => {
  const { asFragment } = render(
    <Divider direction="vertical" variant="dashed" />,
  )
  expect(asFragment()).toMatchSnapshot()
})

test("Divider renders with colorScheme", () => {
  const { asFragment } = render(
    <Divider direction="vertical" colorScheme={"orange"} />,
  )
  expect(asFragment()).toMatchSnapshot()
})

test("Divider renders with text", () => {
  const { asFragment } = render(
    <div>
      <Divider direction="vertical" text="this is a divider" textSize="20px" />
      <Divider
        direction="vertical"
        text="this is a divider"
        textAlign="start"
      />
      <Divider direction="vertical" text="this is a divider" textAlign="end" />
    </div>,
  )
  expect(asFragment()).toMatchSnapshot()
})
