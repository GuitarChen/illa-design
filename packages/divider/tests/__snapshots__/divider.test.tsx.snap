// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider renders default 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #e5e6eb;
}

.emotion-0 span {
  padding: 0 16px;
}

.emotion-1 {
  vertical-align: middle;
  border-style: solid;
  border-width: 0 0 1px 0;
  width: 100%;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    />
  </div>
</DocumentFragment>
`;

exports[`Divider renders horizontal 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #e5e6eb;
}

.emotion-0 span {
  padding: 0 16px;
}

.emotion-1 {
  vertical-align: middle;
  border-style: solid;
  border-width: 0 0 1px 0;
  width: 100%;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    />
  </div>
</DocumentFragment>
`;

exports[`Divider renders vertical 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #e5e6eb;
}

.emotion-0 span {
  padding: 0 16px;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  border-width: 0 0 0 1px;
  border-image: initial;
  border-style: dashed;
  height: 1em;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    />
  </div>
</DocumentFragment>
`;

exports[`Divider renders with colorScheme 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 span {
  padding: 0 16px;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  border-width: 0 0 0 1px;
  border-image: initial;
  border-style: solid;
  height: 1em;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    />
  </div>
</DocumentFragment>
`;

exports[`Divider renders with text 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #e5e6eb;
}

.emotion-0 span {
  padding: 0 16px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #e5e6eb;
  font-size: 20px;
}

.emotion-1 span {
  padding: 0 16px;
}

.emotion-2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-basis: 24px;
  -ms-flex-preferred-size: 24px;
  flex-basis: 24px;
  vertical-align: middle;
  border-style: solid;
  border-width: 0 0 1px 0;
  width: 100%;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #e5e6eb;
  font-size: 14px;
}

.emotion-5 span {
  padding: 0 16px;
}

.emotion-6 {
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: 24px;
  -ms-flex-preferred-size: 24px;
  flex-basis: 24px;
  vertical-align: middle;
  border-style: solid;
  border-width: 0 0 1px 0;
  width: 100%;
}

<div>
    <div
      class="emotion-0"
    >
      <div
        class="emotion-1"
        direction="vertical"
      >
        <div
          class="emotion-2"
        />
        <span>
          this is a divider
        </span>
        <div
          class="emotion-2"
        />
      </div>
    </div>
    <div
      class="emotion-0"
    >
      <div
        class="emotion-5"
        direction="vertical"
      >
        <div
          class="emotion-6"
        />
        <span>
          this is a divider
        </span>
        <div
          class="emotion-2"
        />
      </div>
    </div>
    <div
      class="emotion-0"
    >
      <div
        class="emotion-5"
        direction="vertical"
      >
        <div
          class="emotion-2"
        />
        <span>
          this is a divider
        </span>
        <div
          class="emotion-6"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
